### ISSUE TITLE
Bad key override in Chef::NodeMap filter expression

### ISSUE THREAD

**Issue by bcg62** [jssjr/git#72]
Getting this with new version 4.2.4

```
Compiling Cookbooks...

================================================================================
Recipe Compile Error in /var/chef/cache/cookbooks/git/libraries/provider_git_client_package.rb
================================================================================

RuntimeError
------------
Bad key override in Chef::NodeMap filter expression

Cookbook Trace:
---------------
  /var/chef/cache/cookbooks/git/libraries/provider_git_client_package.rb:7:in `<class:Package>'
  /var/chef/cache/cookbooks/git/libraries/provider_git_client_package.rb:4:in `<class:GitClient>'
  /var/chef/cache/cookbooks/git/libraries/provider_git_client_package.rb:3:in `<class:Provider>'
  /var/chef/cache/cookbooks/git/libraries/provider_git_client_package.rb:2:in `<class:Chef>'
  /var/chef/cache/cookbooks/git/libraries/provider_git_client_package.rb:1:in `<top (required)>'

Relevant File Content:
----------------------
/var/chef/cache/cookbooks/git/libraries/provider_git_client_package.rb:

  1:  class Chef
  2:    class Provider
  3:      class GitClient
  4:        class Package < Chef::Provider::GitClient
  5:          include Chef::DSL::IncludeRecipe
  6:
  7>>         provides :git_client, os: 'linux', override: true if respond_to?(:provides)
  8:
  9:          action :install do
 10:            # FIXME: rhel 5
 11:            include_recipe 'yum-epel' if node['platform_family'] == 'rhel' && node['platform_version'].to_i == 5
 12:
 13:            # Software installation
 14:            package "#{new_resource.name} :create #{parsed_package_name}" do
 15:              package_name parsed_package_name
 16:              version parsed_package_version


Running handlers:
Running handlers complete
Chef Client failed. 0 resources updated in 4.953087 seconds
```

**Comment by SqiSch** [comment]
me too: 

```
================================================================================ESC[0m
ESC[31mRecipe Compile Error in /var/chef/cache/cookbooks/git/libraries/provider_git_client_package.rbESC[0m
================================================================================ESC[0m

ESC[0mRuntimeErrorESC[0m
------------ESC[0m
Bad key override in Chef::NodeMap filter expressionESC[0m

ESC[0mCookbook Trace:ESC[0m
---------------ESC[0m
  /var/chef/cache/cookbooks/git/libraries/provider_git_client_package.rb:7:in `<class:Package>'
ESC[0m  /var/chef/cache/cookbooks/git/libraries/provider_git_client_package.rb:4:in `<class:GitClient>'
ESC[0m  /var/chef/cache/cookbooks/git/libraries/provider_git_client_package.rb:3:in `<class:Provider>'
ESC[0m  /var/chef/cache/cookbooks/git/libraries/provider_git_client_package.rb:2:in `<class:Chef>'
ESC[0m  /var/chef/cache/cookbooks/git/libraries/provider_git_client_package.rb:1:in `<top (required)>'ESC[0m

ESC[0mRelevant File Content:ESC[0m
----------------------ESC[0m
/var/chef/cache/cookbooks/git/libraries/provider_git_client_package.rb:
ESC[0m
ESC[0m  1:  class Chef
ESC[0m  2:    class Provider
ESC[0m  3:      class GitClient
ESC[0m  4:        class Package < Chef::Provider::GitClient
ESC[0m  5:          include Chef::DSL::IncludeRecipe
ESC[0m  6:  
ESC[0m  7>>         provides :git_client, os: 'linux', override: true if respond_to?(:provides)
ESC[0m  8:  
ESC[0m  9:          action :install do
ESC[0m 10:            # FIXME: rhel 5
ESC[0m 11:            include_recipe 'yum-epel' if node['platform_family'] == 'rhel' && node['platform_version'].to_i == 5
ESC[0m 12:  
ESC[0m 13:            # Software installation
ESC[0m 14:            package "#{new_resource.name} :create #{parsed_package_name}" do
ESC[0m 15:              package_name parsed_package_name
ESC[0m 16:              version parsed_package_version

```

**Comment by MiguelMoll** [comment]
Seeing the same issue here.

**Comment by wincus** [comment]
Yeah, me too. Is there any workaround for this?

**Comment by ermaker** [comment]
Me, too.

**Comment by someara** [comment]
Investigating... Chef 11.x?

**Comment by ermaker** [comment]
My information is here.

$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.1 LTS
Release:        14.04
Codename:       trusty
$ chef-solo -v
Chef: 12.3.0

**Comment by MiguelMoll** [comment]
vagrant@default-invision-1404:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.1 LTS
Release:        14.04
Codename:       trusty
-------
chef-solo -v
Chef: 12.4.1

**Comment by brianbianco** [comment]
having this same issue:

$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.2 LTS
Release:	14.04
Codename:	trusty

$ chef-client --version
Chef: 12.0.3


================================================================================
Recipe Compile Error in /.chef/local-mode-cache/cache/cookbooks/git/libraries/provider_git_client_package.rb
================================================================================

RuntimeError
------------
Bad key override in Chef::NodeMap filter expression

**Comment by someara** [comment]
This should be fixed up in 4.3.0

**Comment by docwhat** [comment]
Unfortunately, 4.3.0 has bug #73.  The last working version, if you use windows, is 4.2.2.
